### PR TITLE
New test for checking if ICache backup operations apply default expiry policy to backup entries

### PR DESCRIPTION
**Pull request by yozank**
Test for hazelcast/hazelcast-enterprise#2641.

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java [hunk 1]
```diff
@@ -284,6 +284,49 @@ public void test_whenEntryIsUpdatedBackupIsNotCleaned() {
...
+        ICache cache = createCache(cacheConfig);
+
+        for (int i = 0; i < 100; i++) {
+            cache.put(i, i);
+        }
+
+        sleepAtLeastMillis(500); // make sure async backup operations are completed.
+
+        // Check if all backup entries have applied the default expiry policy
+        for (int i = 1; i < CLUSTER_SIZE; i++) {
```
[the remark this answers is not in the document]
Comment (yozank) [reply hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java]: fixed

File: hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java [hunk 2]
```diff
@@ -284,6 +284,49 @@ public void test_whenEntryIsUpdatedBackupIsNotCleaned() {
...
+                TestBackupUtils.assertExpirationTimeExists(j, backupAccessor);
+            }
+        }
+
+        // terminate 2 nodes to cause backup promotion at the 0th member
+        getNode(instances[1]).shutdown(true);
+        getNode(instances[2]).shutdown(true);
+
+        // expiration time is over.
+        sleepAtLeastMillis(4500);
```
[the remark this answers is not in the document]
Comment (yozank) [reply hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java]: done

File: hazelcast/src/test/java/com/hazelcast/test/backup/TestBackupUtils.java [hunk 3]
```diff
@@ -142,4 +143,23 @@ public void run() {
...
         });
     }
+
+    public static <K, V> void assertExpirationTimeExistsEventually(final K key, final BackupAccessor<K, V> accessor) {
+
+        checkInstanceOf(CacheBackupAccessor.class, accessor, "Need to supply a CacheBackupAccessor");
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertTrueEventually(new AssertTask() {
```
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/test/backup/TestBackupUtils.java]: this `assertTrueEventually` is inside another `assertTrueEventually`. We can remove the one inside.

File: hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java [hunk 4]
```diff
@@ -284,6 +284,47 @@ public void test_whenEntryIsUpdatedBackupIsNotCleaned() {
         }
     }
 
+    @Test
+    public void test_backupOperationAppliesDefaultExpiryPolicy() {
+
```
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java]: empty line

File: hazelcast/src/test/java/com/hazelcast/test/backup/TestBackupUtils.java (hunk 3 above)
Comment (yozank) [reply hazelcast/src/test/java/com/hazelcast/test/backup/TestBackupUtils.java]: eventual inception... fixed

File: hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java (hunk 4 above)
Comment (yozank) [reply hazelcast/src/test/java/com/hazelcast/cache/eviction/CacheExpirationTest.java]: fixed.

**Comment by yozank** [comment]
run-lab-run

**Pull request merged**